databox/core/lib/ServerHooks.ts

```diff
@@ -0,0 +1,132 @@
+    this.databoxesDir = env.databoxStorage;
+    for (const runner of Object.values(this.runnersByModuleName)) {
+      await runner.start(this.databoxesDir);
+    }
+  }
+
+  public static async close(): Promise<void> {
+    this.packageRegistry?.flush();
+    for (const runner of Object.values(this.runnersByModuleName)) {
+      await runner.close();
+    }
+    this.runnersByModuleName = {};
+
+    for (const connection of this.connections) {
+      await connection.disconnect();
+    }
+    this.connections.clear();
+  }
+
+  public static async upload(databoxPackage: IDataboxPackage): Promise<void> {
+    await this.getPackageRegistry().save(databoxPackage);
+  }
+
+  public static async runLocalFile(scriptHash: string, input?: any): Promise<{ output: any }> {
+    const databox = await this.getPackageRegistry().getByHash(scriptHash);
+    const runner = this.runnersByModuleName[databox.module];
+    if (!runner) {
+      throw new Error(`Server does not support required databox module: ${databox.module}`);
+    }
+    if (!runner.canSatisfyVersion(databox.moduleVersion)) {
+      throw new Error(
+        `The current version of ${databox.module} (${runner.runsDataboxModuleVersion}) is incompatible with this Databox version (${databox.moduleVersion})`,
+      );
+    }
+
+    const manifest = <IDataboxManifest>{
+      scriptEntrypoint: databox.scriptEntrypoint,
+      scriptRollupHash: databox.scriptHash,
+      databoxModuleVersion: databox.moduleVersion,
+      databoxModule: databox.module,
+    };
+
+    return await runner.run(databox.path, manifest, input);
+  }
+
+  public static async run(scriptHash: string, input?: any): Promise<{ output: any }> {
+    const databox = await this.getPackageRegistry().getByHash(scriptHash);
+    const runner = this.runnersByModuleName[databox.module];
+    if (!runner) {
+      throw new Error(`Server does not support required databox module: ${databox.module}`);
+    }
+    if (!runner.canSatisfyVersion(databox.moduleVersion)) {
+      throw new Error(
+        `The current version of ${databox.module} (${runner.runsDataboxModuleVersion}) is incompatible with this Databox version (${databox.moduleVersion})`,
+      );
+    }
+
+    const manifest = <IDataboxManifest>{
+      scriptEntrypoint: databox.scriptEntrypoint,
+      scriptRollupHash: databox.scriptHash,
+      databoxModuleVersion: databox.moduleVersion,
+      databoxModule: databox.module,
+    };
+
+    return await runner.run(databox.path, manifest, input);
+  }
+
+  public static async runLocalScript(scriptPath: string, input?: any): Promise<{ output: any }> {
+    const databoxProcess = new LocalDataboxProcess(scriptPath);
+    const databoxModule = await databoxProcess.fetchModule();
+    const runner = this.runnersByModuleName[databoxModule];
+    if (!runner) {
+      throw new Error(`Server does not support required databox module: ${databoxModule}`);
+    }
+    // TODO: should we extract version from file?
```

Comment:
Seems like we have quite a few TODOs in the code here. Not sure if this is intentional?